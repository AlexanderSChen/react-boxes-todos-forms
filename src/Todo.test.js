import React from "react";
import { render, fireEvent} from "@testing-library/react";
import Todo from "./Todo";

test("Smoke Test: Renders without crashing", function() {
    render(<Todo />);
});

test("Snapshot: Matches", function() {
    const { asFragment } = render(<Todo />);
    expect(asFragment()).toMatchSnapshot();
});

test("Matches snapshot when editing", function() {
    const { asFragment, getByText } = render(<Todo />);
    const editButton = getByText("Edit");
    fireEvent.click(editButton);
    expect(asFragment()).toMatchSnapshot();
});

test("Runs the update function on form submit", function() {
    const updateMock = jest.fn();
    const { getByText } = render(<Todo update={updateMock} />);
    const editButton = getByText("Edit");
    fireEvent.click(editButton);
    const updateButton = getByText("Update");
    fireEvent.click(updateButton);
    expect(updateMock).toHaveBeenCalled();
});

test("Runs the delete function on button click", function() {
    const removeMock = jest.fn();
    const { getByText } = render(<Todo remove={removeMock} />);
    const deleteButton = getByText("X");
    fireEvent.click(deleteButton);
    expect(removeMock).toHaveBeenCalled();
});