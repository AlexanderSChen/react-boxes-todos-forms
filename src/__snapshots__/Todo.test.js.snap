// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot when editing 1`] = `
<DocumentFragment>
  <div>
    <form>
      <input
        type="text"
        value="Default todo"
      />
      <button>
        Update
      </button>
    </form>
  </div>
</DocumentFragment>
`;

exports[`Snapshot: Matches 1`] = `
<DocumentFragment>
  <div>
    <li>
      Default todo
    </li>
    <button>
      Edit
    </button>
    <button>
      X
    </button>
  </div>
</DocumentFragment>
`;
